src/components/Tables/Table.tsx

```diff
@@ -199,7 +198,11 @@ const defaultColumnStyleOptions = {
     maxWidth: "20rem",
 } as const;
 
-const Table = <Data,>({
+const Table = <
+    Data,
+    PaginationType extends PaginationTypeEnum,
+    DbData extends Record<string, any> = {},
```

Comment:
we only need DbData if PaginationType is PaginationType.Server, but I'm not sure if its possible to specify that :/ let me know your thoughts

Reply:
I like the second solution, have done